python/src/scipp/compat/mantid.py

```diff
@@ -771,6 +771,16 @@ def convert_TableWorkspace_to_dataset(ws, error_connection=None, **ignored):
     return dataset
 
 
+def convert_WorkspaceGroup_to_dataarray_dict(group_workspace, **kwargs):
+    workspace_dict = {}
+    for i in range(group_workspace.getNumberOfEntries()):
+        workspace = group_workspace.getItem(i)
+        workspace_name = workspace.name().replace(group_workspace.name(), '')
+        workspace_dict[workspace_name] = from_mantid(workspace, **kwargs)
```

Comment:
Good, so this can in principle handle the fact that the WorkspaceGroup is allowed to hold any number of types of workspace.